endstate_correction/neq.py

```diff
@@ -49,23 +49,21 @@ def perform_switching(
         if save_trajs:
             # if switching trajectories need to be saved, create an empty list at the beginning
             # of each switch for saving conformations
-            switching_trajectory = [] 
+            switching_trajectory = []
 
         # select a random sample
-        x = (
-            np.array(random.choice(samples).value_in_unit(distance_unit))
-            * distance_unit
-        )
+        random_frame = random.randint(0, len(samples.xyz)-1)
+        x = samples.xyz[random_frame]
+        if samples.unitcell_lengths is not None:
+            box_length = samples.unitcell_lengths[x]
```

Comment:
It is a bit confusing here as to what this `x` is.
From `x = samples.xyz[random_frame]`, I would imagine that this x would be the coordinate of frame `random_frame` extracted from `samples.xyz`.
Then I would imagine that the box dimension of frame `random_frame` would be `samples.unitcell_lengths[random_frame]`, but this is not the case, so it might be better to explain what is this `x`?

Reply:
oh, that is bug :-) good catch!
it is exactly as you assumed, it should read `samples.unitcell_lenghts[random_frame]`

Reply:
has been adressed here:
https://github.com/wiederm/endstate_correction/commit/bc21f74810bc0a8fd47b43979d39a7241144e164